src/client/config.rs

```diff
@@ -0,0 +1,46 @@
+use crate::ejudge;
+use crate::errors;
+
+use async_trait::async_trait;
+use errors::Result;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum ClientConfig {
+    Ejudge(<ejudge::EjudgeClient as ConfigClient>::Config),
+}
+
+pub trait AsClientConfig {
+    fn as_client_config(self) -> ClientConfig;
+}
+
+impl AsClientConfig for <ejudge::EjudgeClient as ConfigClient>::Config {
```

Comment:
Seems a bit annoying, isn't it? Do we need a static structure here, instead of an easily scalable dynamic one? Let's investigate other options a bit.

Reply:
Yeap, but it's ok for now. Let's add an issue for extracting all code dependent on concrete judge systems into a single file (probably, `main.rs`).